Framework/API/src/NearestNeighbours.cpp

```diff
@@ -37,9 +23,9 @@ NearestNeighbours::NearestNeighbours(
  * be ignored.
  */
 NearestNeighbours::NearestNeighbours(
-    int nNeighbours, boost::shared_ptr<const Instrument> instrument,
-    const ISpectrumDetectorMapping &spectraMap, bool ignoreMaskedDetectors)
-    : m_instrument(instrument), m_spectraMap(spectraMap),
+    int nNeighbours, const SpectrumInfo &spectrumInfo,
+    const std::vector<specnum_t> spectrumNumbers, bool ignoreMaskedDetectors)
+    : m_spectrumInfo(spectrumInfo), m_spectrumNumbers(spectrumNumbers),
```

Comment:
This is only used from within `NearestNeighbourInfo`. Unless another thread modifies the workspace at the same time (which will lead to all kinds of other problems as well), it should be fine.